server.go

```diff
@@ -45,6 +53,9 @@ func NewServer(port int) (server *Server, err error) {
 	server.Log = cue.NewLogger(server.Id)
 	server.Log.Infof("new server on port %d", port)
 
+	server.listenConfig = &net.ListenConfig{KeepAlive: keepAlive}
+	server.ctx, server.cancelFunc = context.WithCancel(context.Background())
```

Comment:
since we switched to https://golang.org/pkg/net/#ListenConfig.Listen (for keepalive) it make sense to send it correct context with cancellation. And cancel it on stopping the server